File: merge_elevation.py

import json
import subprocess

# 1. Get current file content
with open('/Users/murakamidaisuke/Downloads/Antigraviti/mapbox/src/data/course_sasayama.json', 'r', encoding='utf-8') as f:
    current_file = json.load(f)

# 2. Get old file content from git e85764a
old_file_content = subprocess.check_output(['git', 'show', 'e85764a:src/data/course_sasayama.json'], cwd='/Users/murakamidaisuke/Downloads/Antigraviti/mapbox').decode('utf-8')
old_file = json.loads(old_file_content)

# 3. Extract the elevation-rich LineString coordinates
old_route_feature = next(f for f in old_file['features'] if f['geometry']['type'] == 'LineString')
print(f"Old coordinates count: {len(old_route_feature['geometry']['coordinates'])}")

# 4. Update the current file's LineString
current_route_feature = next(f for f in current_file['features'] if f['geometry']['type'] == 'LineString')
current_route_feature['geometry']['coordinates'] = old_route_feature['geometry']['coordinates']
print(f"New coordinates count: {len(current_route_feature['geometry']['coordinates'])}")

# 5. Write back to file
with open('/Users/murakamidaisuke/Downloads/Antigraviti/mapbox/src/data/course_sasayama.json', 'w', encoding='utf-8') as f:
    json.dump(current_file, f, ensure_ascii=False, indent=4)

print("Successfully merged elevation data.")

File: merge_elevation.js

const fs = require('fs');
const { execSync } = require('child_process');

// 1. Get current file content (with properties and checkpoints)
const currentFile = JSON.parse(fs.readFileSync('/Users/murakamidaisuke/Downloads/Antigraviti/mapbox/src/data/course_sasayama.json', 'utf8'));

// 2. Get old file content (with elevation coordinates) from git e85764a
const oldFileContent = execSync('git show e85764a:src/data/course_sasayama.json').toString();
const oldFile = JSON.parse(oldFileContent);

// 3. Extract the elevation-rich LineString coordinates from the old file
const oldRouteFeature = oldFile.features.find(f => f.geometry.type === 'LineString');
console.log('Old coordinates count:', oldRouteFeature.geometry.coordinates.length);

// 4. Update the current file's LineString with the old coordinates
const currentRouteFeature = currentFile.features.find(f => f.geometry.type === 'LineString');
currentRouteFeature.geometry.coordinates = oldRouteFeature.geometry.coordinates;
console.log('New coordinates count:', currentRouteFeature.geometry.coordinates.length);

// 5. Write back to file
fs.writeFileSync('/Users/murakamidaisuke/Downloads/Antigraviti/mapbox/src/data/course_sasayama.json', JSON.stringify(currentFile, null, 4));
console.log('Successfully merged elevation data.');
